broker.go

```diff
@@ -84,6 +99,23 @@ func (b *Broker) Open(conf *Config) error {
 
 		b.conf = conf
 
+		// Create or reuse the metrics shared between brokers
+		b.incomingByteRate = metrics.GetOrRegisterMeter("incoming-byte-rate", conf.MetricRegistry)
+		b.requestRate = metrics.GetOrRegisterMeter("request-rate", conf.MetricRegistry)
+		b.requestSize = getOrRegisterHistogram("request-size", conf.MetricRegistry)
+		b.outgoingByteRate = metrics.GetOrRegisterMeter("outgoing-byte-rate", conf.MetricRegistry)
+		b.responseRate = metrics.GetOrRegisterMeter("response-rate", conf.MetricRegistry)
+		b.responseSize = getOrRegisterHistogram("response-size", conf.MetricRegistry)
+		// Do not gather metrics for seeded broker (-1 id)
+		if b.id >= 0 {
```

Comment:
Sounds like java uses the ID because IPv6 addresses contain characters that aren't permitted in jmx metric names. I'm OK with following them on this one.

I don't know what limitations this metrics package has, but if it supports all the necessary names then I'm also OK actually using the address:port combo instead of the ID.